src/applications/vaos/reducers/newAppointment.js

```diff
@@ -393,9 +393,34 @@ export default function formReducer(state = initialState, action) {
         },
       };
     }
-    case FORM_REASON_FOR_APPOINTMENT_UPDATE_REMAINING_CHAR: {
+    case FORM_REASON_FOR_APPOINTMENT_CHANGED: {
+      let newSchema = state.pages.reasonForAppointment;
+
+      // Update additional info title based on radio selection
+      const additionalInfoTitle =
+        action.data.reasonForAppointment === 'other'
+          ? REASON_ADDITIONAL_INFO_TITLES.other
+          : REASON_ADDITIONAL_INFO_TITLES.default;
+
+      newSchema = set(
```

Comment:
I think I mentioned in Slack that the other way to do this is an `updateSchema` property in `ui:options` on the `uiSchema` object. You're forced to do that when using our big form configs, but we have extra flexibility and I prefer doing it the way you have it here on our pages.